core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -293,19 +339,15 @@ public JsonArray getAdapterConfigurations() {
      * @return The configuration properties or {@code null} if no specific
      *         properties have been set.
      * @throws NullPointerException if type is {@code null}.
+     * @throws IllegalArgumentException if the conversion of the adapter object fails.
```

Comment:
when/why would that happen? We have explicitly taken care that the `Adapter` class can be mapped to JSON, haven't we?

Reply:
It should not happen as we already take care with the `Adapter` class. I will remove this from the _javadoc_.